cmd/collectors/rest/plugins/certificate/certificate.go

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright NetApp Inc, 2022 All rights reserved
+ */
+
+package certificate
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"github.com/tidwall/gjson"
+	"goharvest2/cmd/poller/plugin"
+	"goharvest2/cmd/tools/rest"
+	"goharvest2/pkg/conf"
+	"goharvest2/pkg/dict"
+	"goharvest2/pkg/errors"
+	"goharvest2/pkg/logging"
+	"goharvest2/pkg/matrix"
+	"goharvest2/pkg/tree/node"
+	"time"
+)
+
+const DefaultPluginDuration = 30 * time.Minute
+const DefaultDataPollDuration = 3 * time.Minute
+
+type Certificate struct {
+	*plugin.AbstractPlugin
+	data                 *matrix.Matrix
+	instanceKeys         map[string]string
+	instanceLabels       map[string]*dict.Dict
+	pluginInvocationRate int
+	currentVal           int
+	client               *rest.Client
+	query                string
+}
+
+func New(p *plugin.AbstractPlugin) plugin.Plugin {
+	return &Certificate{AbstractPlugin: p}
+}
+
+func (my *Certificate) Init() error {
+
+	var err error
+
+	if err = my.InitAbc(); err != nil {
+		return err
+	}
+
+	timeout := rest.DefaultTimeout * time.Second
+	if my.client, err = rest.New(conf.ZapiPoller(my.ParentParams), timeout); err != nil {
+		my.Logger.Error().Stack().Err(err).Msg("connecting")
+		return err
+	}
+
+	if err = my.client.Init(5); err != nil {
+		return err
+	}
+
+	// Assigned the value to currentVal so that plugin would be invoked first time to populate cache.
+	if my.currentVal, err = my.setPluginInterval(); err != nil {
+		my.Logger.Error().Err(err).Stack().Msg("Failed while setting the plugin interval")
+		return err
+	}
+
+	return nil
+}
+
+// For reference: https://opengrok-prd.eng.netapp.com/source/xref/bard_main/modules/dfm-data-access/src/main/java/com/netapp/dfm/entity/platform/event/builtin/ClusterSelfSignedCertificateCheckListener.java#67
+func (my *Certificate) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
+
+	if my.currentVal >= my.pluginInvocationRate {
+		my.currentVal = 0
+
+		// invoke private vserver cli rest and get admin vserver name
+		if adminVserver, err := my.GetAdminVserver(); err != nil {
+			my.Logger.Warn().Stack().Err(err).Msg("Failed to collect admin vserver")
+			return nil, nil
+		} else {
+			// invoke private ssl cli rest and get admin vserver's serial number
+			if adminVserverSerial, err := my.GetSecuritySsl(adminVserver); err != nil {
+				my.Logger.Warn().Stack().Err(err).Msg("Failed to collect admin vserver's serial number")
+				return nil, nil
+			} else {
```

Comment:
minor: this else clause isn't needed since if err != nil we return

Reply:
done